src/common/PhysicsController.cpp

```diff
@@ -0,0 +1,55 @@
+#include <vector>
+#include <memory>
+#include <iostream>
+
+#include "PhysicsController.hpp"
+
+#include "../common/util.hpp"
+
+PhysicsController::PhysicsController() {}
+
+std::shared_ptr<CircleRigidBody> PhysicsController::createCRB(float radius, sf::Vector2f position) {
+    std::shared_ptr<CircleRigidBody> crb = std::shared_ptr<CircleRigidBody>(
+        new CircleRigidBody(radius, position));
+    mCircleRigidBodies.push_back(crb);
+    return crb;
+}
+
+void PhysicsController::step() {
+    for (auto crb : mCircleRigidBodies) {
+        crb->step();
```

Comment:
is it important that the steps happen "at once" I  can imagine half the circles being updated right when the frame updates or something. probably not a real issue, just thinking out loud. safe to ignore

Reply:
We set the network state _after_ we've updated the game state. The physics controller step is a substep of "update game state". 
What you're thinking about isn't a concern.